### PR TITLE
done

### PR DESCRIPTION
**Pull request by walamana**

### CODE REVIEW

File: src/app/about/page.tsx [hunk 1]
```diff
@@ -0,0 +1,5 @@
+"use client"
```
Comment (uwumarogie) [inline src/app/about/page.tsx:1]: page.tsx file are most of the server components. meaning you don't need the "use client"

File: src/app/layout.tsx [hunk 2]
```diff
@@ -16,7 +17,22 @@ export default function RootLayout({
...
+          <span className="font-sans text-lg">Header</span>
+        </div>
+        <div className="col-span-1 p-5 bg-slate-950 h-full flex flex-col">
+          <Link className="p-3" href="/">Home</Link>
+          <Link className="p-3" href="/about">About</Link>
+          <Link className="p-3" href="/products">Products</Link>
+          <Link className="p-3" href="/contact">Contact</Link>
+          <Link className="p-3" href="/search">Search</Link>
+          <Link className="p-3" href="/docs">Docs</Link>
+        </div>
```
Comment (uwumarogie) [inline src/app/layout.tsx:31]: noice idea. just place it in a separate function

File: src/app/search/layout.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+"use client"
+
+import {useState} from "react";
+import {useRouter} from "next/navigation";
+
+export default function SearchLayout({children}) {
```
Comment (uwumarogie) [inline src/app/search/layout.tsx:6]: ({children}: {children: React.ReactNode})

File: src/app/search/layout.tsx [hunk 4]
```diff
@@ -0,0 +1,29 @@
+"use client"
```
Comment (uwumarogie) [inline src/app/search/layout.tsx:1]: why do you need here a "use client"? It's correct. But why

File: src/app/search/results/page.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+        </div>
+        <div className="mb-12">
+            <span className="text-sm opacity-60">Results from api.duckduckgo.com</span>
+        </div>
+        <div>
+            {result?.Image && <img src={"https://duckduckgo.com" + result?.Image} className="h-32 mb-4"/>}
+            <h1 className="text-4xl mb-4">{result?.Heading}</h1>
+            <span>{result?.AbstractText}</span>
+            <p className="text-sm opacity-60">From <a href={result?.AbstractURL}>{result?.AbstractSource}</a></p>
+        </div>
```
Comment (uwumarogie) [inline src/app/search/results/page.tsx:29]: search for the workflow of server component to client component. Tipp: how don't need the useEffect :)

File: src/app/search/results/page.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    const query = useSearchParams<{term: string}>()
+    const [result, setResult] = useState(null)
+    useEffect(() => {
+        fetch("https://api.duckduckgo.com/?q=" + query.get("term") + "&format=json&atb=v380-1")
+            .then(res => res.json())
+            .then(res => {
+                setResult(res)
+            })
+    }, [query.get("term")])
+    return <div>
```
Comment (uwumarogie) [inline src/app/search/results/page.tsx:17]: please add some ()

**Review by uwumarogie — CHANGES_REQUESTED** [review]

File: src/app/search/layout.tsx (hunk 4 above)
Comment (walamana) [reply src/app/search/layout.tsx:1]: Don't have the code on hand but I think there is a request that uses client api. I could possibly do it with some sort of server request. But I was to lazy to look that up right now 😅

Comment (walamana) [reply src/app/search/layout.tsx:1]: Oh nice, I just looked it up and saw that fetch is also available server side. I'll have a look at it some time :)

Comment (uwumarogie) [reply src/app/search/layout.tsx:1]: Everytime you have interactivity in the component e.g. by clicking a button when you use useState you have to use "use client". 

workflow: server to client component

**Pull request closed without merging**